polaris/polaris/management/commands/watch_transactions.py

```diff
@@ -185,13 +185,23 @@ def find_matching_payment_op(
                 matching_payment_op = operation
                 break
 
-        return matching_payment_op
+        return PaymentOp.from_xdr(matching_payment_op.to_xdr_object().to_xdr())
 
-    @staticmethod
-    def _check_payment_op(operation: Operation, want_asset: Asset) -> bool:
+    @classmethod
+    def _check_payment_op(cls, generic_operation: Operation, want_asset: Asset) -> bool:
+        if generic_operation.TYPE != OperationType.PAYMENT:
+            return False
+        payment_operation = PaymentOp.from_xdr(
+            generic_operation.to_xdr_object().to_xdr()
+        )
+        asset_num_x = (
+            payment_operation.asset.alpha_num4 or payment_operation.asset.alpha_num12
+        )
+        asset_code = (
+            asset_num_x.asset_code.asset_code4 or asset_num_x.asset_code.asset_code12
+        ).decode()
```

Comment:
@overcat `asset_code4` and `asset_code12` are `bytes`, will decoding them give me the string representation or an XDR string?